src/main/java/com/bean00/Server.java

```diff
@@ -11,12 +10,18 @@ public Server(BufferedReader in, PrintWriter out) {
         messageController = new MessageController(in, out);
     }
 
-    public void handleGET() throws IOException {
-        Request request = messageController.readRequest();
+    public void run() {
+        try {
+            Request request = messageController.createRequest();
 
-        Response response = messageController.interpretRequest(request);
+            Response response = messageController.interpretRequest(request);
 
-        messageController.writeResponse(response);
+            messageController.writeResponse(response);
+        } catch (Throwable t) {
+            Response response = new Response(500);
```

Comment:
Soon, you should be addressing this magic number problem.

Reply:
I've created a Status class that has constants and a map of the status codes to messages. However, I'm not sure if I should leave the hard-coded integers in the tests, or use the constants in the tests as well (the constants are already being used in the regular classes).